controllers/userController.js

```diff
@@ -1,90 +1,348 @@
+    const email = req.body.email;
+   
+    let verify = await this.verifyUser( email );
+
+    if (verify.NUM == 0) {
+      return res.send({ status: STATUS_CODE.INCORRECT_USER_EMAIL, result: email });
+    }
+
+    let result = await this.removeUser( email );
+
+    if (result.affectedRows){
+      return res.send({ status: STATUS_CODE.SUCCESS });
+    }
+  } catch (error) {
+    return res.send({ status: STATUS_CODE.ERROR, result: error });
+  }
+  console.log('success');
+  return res.send({ status: STATUS_CODE.SUCCESS});
+}
+
+exports.login = async (req, res) => {
+  const query = `SELECT * FROM users WHERE email = ?`;
+  const {email, password, code, originalCode} = req.body;
+  connection.promise().query(query, [email.toLowerCase()])
+    .then(data => {
+      if (data[0].length === 0) {
+        return res.send({status: STATUS_CODE.INVALID_EMAIL});
+      } else if (data[0][0].status === REGISTRATION_STATUS.UNREGISTERED){
+        return res.send({status: STATUS_CODE.REQUEST_SIGN_UP});
+      } else if (code === originalCode){
+        const inputPassword = hashPassword(password, data[0][0].salt);
+        if (inputPassword === data[0][0].hash){
+          return res.send({ status: STATUS_CODE.SUCCESS});
+        } else {
+          return res.send({status: STATUS_CODE.INVALID_PASSWORD});
+        }
+      } else {
+        return res.send({ status: STATUS_CODE.INVALID_CODE});
+      }
+    })
+    .catch(error => {
+      console.log(error);
+      res.send({status: STATUS_CODE.ERROR});
+    });
+}
+
+exports.sendVerificationEmail = async (req, res) =>{
+  const code = Math.floor(100000 + Math.random() * 900000);
+  const mailOptions = {
+    from: process.env.MAIL_EMAIL,
+    to: req.body.email,
+    subject: 'VUcept Management Verification Code',
+    text: 'Please enter the following verification code: ' + code
+  };
+  transport.sendMail(mailOptions, (error, info) => {
+    if (error) {
+      res.send({status: STATUS_CODE.ERROR});
+    } else {
+      res.send({status: STATUS_CODE.SUCCESS, code: code})
+    }
+  });
+}
 
-  connection.promise().query(query, email)
+exports.signUp = async (req, res) => {
+  const {password, code, originalCode, email} = req.body;
+  //check for verification code
+  if (code !== originalCode) {
+    return res.send({status: STATUS_CODE.INVALID_CODE});
+  }
+  //check for user status 
+  const queryEmail = `SELECT * FROM users WHERE email = ?`;
+  connection.promise().query(queryEmail, [email])
   .then(data => {
-    if (data[0].affectedRows) {
-      message = 'user deleted successfully';
-      res.status(200).send({message: message, data: data[0]});
+    if (data[0].length === 0) {
+      console.log(data[0])
+      return res.send({status: STATUS_CODE.INVALID_EMAIL});
+    } else {
+      //data[0][0] contains the actual json object that gets returned by mysql
+      if (data[0][0].status !== REGISTRATION_STATUS.UNREGISTERED){
+        return res.send({status : STATUS_CODE.USER_EXISTENT});
```

Comment:
I think the log in APIs look great!
Quick question, what does user_existent stand for (lol)?

Reply:
@OliviaLiu417 wrote this part. I think it means the user has already signed up so he or she can't sign up again

Reply:
Yes that means when a user has registered but tries to register again! I believe we should prevent this behavior